### ISSUE TITLE
Support composing 280 character tweet

### ISSUE THREAD

**Issue by IBBoard** [baedert/corebird#780]
Twitter are apparently rolling out 280 character tweets to everyone "soon" ([source](https://www.thedailybeast.com/twitter-expanding-character-count-for-all-users)). As much as it's an awful idea, Corebird should probably support it.

I had heard something about how the "character" counting was a bit odd*, though, so hopefully there is some official documentation.

* Japanese writing getting 140 chars, some characters with multi-byte Unicode values counting as two characters (but not emoji), etc

**Comment by k1l** [comment]
The 280 character limit has just gone live: https://blog.twitter.com/official/en_us/topics/product/2017/tweetingmadeeasier.html

**Comment by Vistaus** [comment]
Yes, thank you, @baedert ! Btw, this would make a nice 1.7.3 release for people who don't use the latest git mater ;)

**Comment by baedert** [comment]
Well, it's not that simple:

> Japanese, Korean, and Chinese will continue to have 140 characters because cramming is not an issue in these languages

So, how would a client be able to tell what "language" a tweet is in? And it doesn't seem to be just that, e.g. I can't send a tweet with 280 emojis here. Works when I use 140 though, while 280 ascii characters just work. The emoji tweet's language doesn't get recognized at all and ends up being undefined (which seems to fall back to 140), if I just type >140 characters of bullshit, the language ends up being recognized as "ind" (indonesian I presume).

**Comment by IBBoard** [comment]
It's a good start, but it's not quite that simple. Apparently some characters count for 1 (Latin-related and "smart" punctuation) and others count for 2 (emoji and Eastern languages).

They seem to be basing it on character points, rather than "language" per se. FakeUnicode is exploring it: https://twitter.com/FakeUnicode/status/928032241153753089

**Comment by Vistaus** [comment]
What a stupid idea from Twitter to count like that. They should've just upped the limit to 180 for all and called it a day... Oh well.

**Comment by IBBoard** [comment]
But that would give Chinese/Japanese/other languages more characters as well (which they apparently don't use anyway because they don't need to), and we can't have that! Much better to make arbitrary rules that they've apparently already added characters to.

Occasionally, FakeUnicode is useful and not just someone who breaks Twitter clients!

**Comment by Vistaus** [comment]
I disagree. They might not use those extra 40 characters, but who's to say *we* are gonna use 280 characters? I know I won't.
